core/src/main/java/org/jruby/RubyNil.java

```diff
@@ -144,7 +144,14 @@ public static RubyFloat to_f(ThreadContext context, IRubyObject recv) {
      */
     @JRubyMethod
     public static RubyString to_s(ThreadContext context, IRubyObject recv) {
-        return RubyString.newEmptyString(context.runtime);
+        Ruby runtime = context.runtime;
+        RubyString emptyString = RubyString.newEmptyString(runtime);
+
+        if (runtime.getInstanceConfig().isFrozenCoreToS()) {
+            return runtime.freezeAndDedupString(emptyString);
```

Comment:
does it make sense to dedup an emty string here ?

Reply:
Is there an issue with deduplication because its cost? Is it a better idea to just return a frozen copy?

Reply:
I was more thinking about of just returning a frozen dedupped singleon for that runtime

Reply:
Right, I see that singleton could be created by the runtime. I just updated the PR accordingly.